### DISCUSSION TITLE
How to mutate q-input's value without it propagating to the parent?

### DISCUSSION THREAD

**Discussion by ivanjaros** [quasarframework/quasar#17286] [General - Components / Directives / etc]
i have q-input for search form. This input will auto-search with delay after user types in the phrase. each change resets timer in watcher and after 350ms it fires.

i am implementing search history as q-menu. the problem i have is that when user picks one of the terms from the history, i want to trigger instant search. but the watcher for the phrase will get triggered as well if i mutate the searched data field. this will cause duplicate search to take place.

i cannot use a flag for the opened history menu to ignore delayed search in the watcher because of how vue ticks work.

i wanted to use the input's model-value on to of v-model and it actually worked except for some reason typing became buggy. the input was not taking input properly, was ignoring keys and deleting content. i guess change/input events were fighting each other.

i also tried to only mutate the html input and not trigger the q-input's v-model but it seems q-input is hooked into the native input's events as well and so this did not work as desired.

q-input has no method i could use to mutate the value without triggering v-model update to bubble up to the parent.

is there some way i could achieve mutation of q-input's value without it propagating to the parent that i have not  mentioned?

--

I have removed the auto-submit logic in favour of user having to press enter, like in the ancient times. But am still interested if this is possible.

**Comment by stefanvanherwijnen** [comment]
Try to use `:model-value=""` instead of `v-model=""` and use `@update:model-value=""` to listen to the update event.

**Reply by ivanjaros** [reply]
yeah, i guess reverting the logic makes sense. i will look at this solution when i will be implementing mobile version.
